CharlieBackend.Api/Controllers/EventsController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using CharlieBackend.Business.Services.Interfaces;
+using CharlieBackend.Core;
+using CharlieBackend.Core.DTO.Schedule;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace CharlieBackend.Api.Controllers
+{
+    /// <summary>
+    /// Controller to manage schedules data
+    /// </summary>
+    [Route("api/events")]
+    [ApiController]
+    public class EventsController : ControllerBase
+    {
+        private readonly IEventsService _eventsService;
+
+
+        /// <summary>
+        /// Events controller constructor
+        /// </summary>
+        /// <param name="scheduleService"></param>
+        public EventsController(IEventsService eventsService)
+        {
+            _eventsService = eventsService;
+        }
+
+        /// <summary>
+        /// Get event by id
+        /// </summary>
+        /// <response code = "200" > Successful getting of schedule</response>
+        /// <response code="HTTP: 404, API: 3">Error, given schedule event not found</response>
+        [SwaggerResponse(200, type: typeof(ScheduledEventDTO))]
+        [Authorize(Roles = "Secretary, Admin")]
+        [HttpGet("{scheduledEventID}")]
+        public async Task<ActionResult<ScheduledEventDTO>> GetConcreteScheduleByID(long scheduledEventID)
+        {
+            var foundScheduleEvent = await _eventsService.GetConcreteScheduleByIdAsync(scheduledEventID);
+
+            return foundScheduleEvent.ToActionResult();
+        }
+
+        /// <summary>
+        /// Updates a single event
+        /// </summary>
+        /// <response code="200">Successful update of schedule</response>
+        /// <response code="HTTP: 404, API: 3">Error, update data is missing</response>
+        /// <response code="HTTP: 400, API: 0">Error, update data is wrong</response>
+        [SwaggerResponse(200, type: typeof(EventOccurrenceDTO))]
+        [Authorize(Roles = "Secretary, Admin")]
+        [HttpPut("{scheduledEventID}")]
+        public async Task<ActionResult<ScheduledEventDTO>> UpdateEventById(long scheduledEventID, [FromBody] UpdateScheduledEventDto request)
```

Comment:
Add Fluent validation to UpdateScheduledEventDto

Reply:
Added. I only did this for UpdateScheduledEventDto, but I think it should be done for updated EventOccurrences as well. If my changes are ok it can be added as a separate issue and pull request.